backend/regalloc/regalloc_irc.ml

```diff
@@ -508,13 +511,12 @@ let run : Cfg_with_infos.t -> Cfg_with_infos.t =
   let spilling_because_unused = Reg.Set.diff cfg_infos.res cfg_infos.arg in
   (match Reg.Set.elements spilling_because_unused with
   | [] -> ()
-  | _ :: _ as spilled_nodes -> (
+  | _ :: _ as spilled_nodes ->
     List.iter spilled_nodes ~f:(fun reg -> State.add_spilled_nodes state reg);
     (* note: rewrite will remove the `spilling` registers from the "spilled"
        work list and set the field to unknown. *)
-    match rewrite state cfg_with_infos ~spilled_nodes ~reset:false with
-    | false -> ()
-    | true -> Cfg_with_infos.invalidate_liveness cfg_with_infos));
+    let (_ : bool) = rewrite state cfg_with_infos ~spilled_nodes ~reset:false in
```

Comment:
The result of `rewrite` is always `false` here I think. Maybe add an `assert` instead of ignoring it?

Reply:
I am not entirely sure; are you suggesting
it is because of `Cfg_deadcode`?

Reply:
sorry, it was my mistake. this code is fine and I think both values are possible.